### PR TITLE
8661 Step 2 cleanup/enhancements

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#8661

*Description of changes:*
- added disabled button + icon style overrides
- added disabled checkbox style overrides
- updated helper text
- added roles v-card
- conditionally hide CP role checkbox
- misc cleanup
- app version: 2.0.13
- created separate template blocks for person vs org
- moved business rules out of template and into script (getters)
- pre-select (lock) Director role if it's the only possibility
- make director's initial delivery address same as mailing address
- fixed unit test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/AddPeopleAndRoles/ListPeopleAndRoles.vue [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <ConfirmRemoveDialog
       :dialog="dialog"
       attach="#people-roles"
-      @confirm="emitRemovePerson(personId)"
+      @confirm="emitRemovePerson(activeIndex)"
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/ListPeopleAndRoles.vue:6]: Just a rename to be consistent with other code.

File: src/components/AddPeopleAndRoles/OrgPerson.vue [hunk 2]
```diff
@@ -46,97 +47,100 @@
...
-                  <v-text-field
-                    filled
-                    class="item"
-                    label="Full Legal Corporation or Firm Name"
-                    id="firm-name"
-                    v-model="orgPerson.officer.orgName"
-                    :rules="orgNameRules"
-                  />
-                </div>
+                <!-- Person Name -->
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/OrgPerson.vue:50]: Go into setting dropdown (gear icon above diffs) and hide whitespace changes to see this more neatly.

File: src/components/AddPeopleAndRoles/OrgPerson.vue [hunk 3]
```diff
@@ -355,7 +422,7 @@ export default class OrgPerson extends Mixins(EntityFilterMixin, CommonMixin) {
       if (this.reassignCompletingParty) {
         this.emitReassignCompletingPartyEvent()
       }
-      const person: OrgPersonIF = this.addPerson()
+      const person = this.setPerson()
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/OrgPerson.vue:425]: Type is already assigned by the method.

File: src/components/AddPeopleAndRoles/OrgPerson.vue [hunk 4]
```diff
@@ -433,14 +498,10 @@ export default class OrgPerson extends Mixins(EntityFilterMixin, CommonMixin) {
     }
   }
 
-  // *** FUTURE: fix or delete this (activeIndex is never -1 so this always returns false)
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/OrgPerson.vue:436]: This was incorrect. The active index is -1 for a new org/person, and >= 0 for editing.

File: src/components/AddPeopleAndRoles/OrgPerson.vue [hunk 5]
```diff
@@ -467,25 +528,15 @@ export default class OrgPerson extends Mixins(EntityFilterMixin, CommonMixin) {
...
   // Event emitters
   @Emit('addEditPerson')
   private emitPersonInfo (personInfo: OrgPersonIF): void {}
 
   @Emit('resetEvent')
   private emitResetEvent (): void {}
 
-  @Emit('removePersonEvent')
-  private emitRemovePersonEvent (activeIndex: number): void {}
+  @Emit('removePerson')
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/OrgPerson.vue:538]: Renamed for consistency with other code.

File: src/components/AddPeopleAndRoles/OrgPerson.vue [hunk 6]
```diff
@@ -606,17 +657,6 @@ li {
   }
 }
 
-.highlighted-role {
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/OrgPerson.vue:609]: This is replaced by the design with the read-only checkbox.

File: src/components/AddPeopleAndRoles/OrgPerson.vue [hunk 7]
```diff
@@ -630,4 +670,8 @@ li {
     color: $gray7;
   }
 }
+
+.gray-card {
+  background-color: rgba(0, 0, 0, 0.06);
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/OrgPerson.vue:675]: Same "colour" as app background.

File: src/components/AddPeopleAndRoles/PeopleAndRoles.vue [hunk 8]
```diff
@@ -70,6 +70,7 @@
         id="btn-start-add-cp"
         outlined
         color="primary"
+        class="btn-outlined-primary"
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/PeopleAndRoles.vue:73]: For CSS selector to show disabled style.

File: src/components/AddPeopleAndRoles/PeopleAndRoles.vue [hunk 9]
```diff
@@ -180,9 +182,8 @@ export default class PeopleAndRoles extends Mixins(EntityFilterMixin) {
   @Action setAddPeopleAndRoleStepValidity!: ActionBindingIF
 
   private helpToggle = false
-  private showOrgPersonForm: boolean = false
-  private activeIndex: number = -1
-  private addEditInProgress: boolean = false
```
Comment (severinbeauvais) [inline src/components/AddPeopleAndRoles/PeopleAndRoles.vue:185]: Obsolete, so I deleted it.

File: src/components/DefineCompany/OfficeAddresses.vue [hunk 10]
```diff
@@ -265,8 +265,8 @@ export default class OfficeAddresses extends Mixins(CommonMixin, EntityFilterMix
   // State of the checkbox for determining whether the Record address is the same as the Registered address
   private inheritRegisteredAddress: boolean = true
 
-  // The Address schema containing Vuelidate rules.
-  private addressSchema = officeAddressSchema
+  // Office Address schema for template
+  readonly OfficeAddressSchema = OfficeAddressSchema
```
Comment (severinbeauvais) [inline src/components/DefineCompany/OfficeAddresses.vue:269]: This is a "const" or read-only thing so it should not be named like a variable.

File: tests/unit/OrgPerson.spec.ts [hunk 11]
```diff
@@ -334,13 +334,18 @@ describe('Org Person component', () => {
     wrapper.destroy()
   })
 
-  it('Shows popup if there is a completing party', async () => {
+  it('Shows popup if there is already a completing party', async () => {
+    store.state.stateModel.tombstone.authRoles = ['staff']
```
Comment (severinbeauvais) [inline tests/unit/OrgPerson.spec.ts:338]: Only staff can reassign the completing party.

**Review by cameron-freshworks — APPROVED** [review]
Looks good!

**Review by andrepestana-aot — COMMENTED** [review]
LGTM

**Pull request merged**